classes/cliTool/CliTool.inc.php

```diff
@@ -72,13 +78,84 @@ function __construct($argv = array()) {
 
 		$this->scriptName = isset($this->argv[0]) ? array_shift($this->argv) : '';
 
+		$this->checkArgsForUsername();
+
 		if (isset($this->argv[0]) && $this->argv[0] == '-h') {
-			$this->usage();
-			exit(0);
+			$this->exitWithUsageMessage();
 		}
 	}
 
 	function usage() {
 	}
 
+	private function checkArgsForUsername() {
+		$usernameKeyPos = array_search('--user_name', $this->argv);
+		if (!$usernameKeyPos) {
+			$usernameKeyPos = array_search('-u', $this->argv);
+		}
+
+		if ($usernameKeyPos) {
+			$usernamePos = $usernameKeyPos + 1;
+			if (count($this->argv) >= $usernamePos + 1) {
+				$this->username = $this->argv[$usernamePos];
+
+				unset($this->argv[$usernamePos]);
+			}
+
+			unset($this->argv[$usernameKeyPos]);
+		}
+
+		$userDao = DAORegistry::getDAO('UserDAO'); /** @var $userDao UserDAO */
+
+		if ($this->username) {
+
+			$user = $userDao->getByUsername($this->username);
+
+			$this->setUser($user);
+		}
+
+		if (!$this->user) {
+			$userGroupDao = DAORegistry::getDAO('UserGroupDAO'); /** @var $userGroupDao UserGroupDAO */
+			$adminGroups = $userGroupDao->getUserGroupIdsByRoleId(ROLE_ID_SITE_ADMIN);
+
+			if (count($adminGroups)) {
+				$groupUsers = $userGroupDao->getUsersById($adminGroups[0])->toArray();
+
+				if (count($groupUsers) > 0) {
+					$this->setUser($groupUsers[0]);
+				} else {
+					$this->exitWithUsageMessage();
+				}
+			}
```

Comment:
If this condition fails: `if (count($adminGroups))`, then no user could be set and the CLI should exit with the usage message, right?

Reply:
@NateWr are you confirming that this is intentional? Even though not all CLI tools need a user to run their code, the initial case that this PR is trying to fix, is the one where the CLI tool code makes, for example, an service call, that in turn needs a user defined in order to be executed. So, this PR tries to make the user mandatory for CLI tools.

Reply:
That's a good point. I guess we shouldn't enforce this yet. So leave it as-is. :+1: